src/sync/remote_mongo_collection.hpp

```diff
@@ -0,0 +1,290 @@
+    /// @param completion_block The resulting json string or error if one occurs
+    void find(const std::string& filter_json,
+              RemoteFindOptions options,
+              std::function<void(std::string, util::Optional<AppError>)> completion_block);
+    
+    /// Finds the documents in this collection which match the provided filter.
+    /// @param filter_json A `Document` as a json string that should match the query.
+    /// @param completion_block The resulting json string or error if one occurs
+    void find(const std::string& filter_json,
+              std::function<void(std::string, util::Optional<AppError>)> completion_block);
+
+    /// Returns one document as a json string from a collection or view which matches the
+    /// provided filter. If multiple documents satisfy the query, this method
+    /// returns the first document according to the query's sort order or natural
+    /// order.
+    /// @param filter_json A `Document` as a json string that should match the query.
+    /// @param options `RemoteFindOptions` to use when executing the command.
+    /// @param completion_block The resulting json string or error if one occurs
+    void find_one(const std::string& filter_json,
+                  RemoteFindOptions options,
+                  std::function<void(std::string, util::Optional<AppError>)> completion_block);
+    
+    /// Returns one document as a json string from a collection or view which matches the
+    /// provided filter. If multiple documents satisfy the query, this method
+    /// returns the first document according to the query's sort order or natural
+    /// order.
+    /// @param filter_json A `Document` as a json string that should match the query.
+    /// @param completion_block The resulting json string or error if one occurs
+    void find_one(const std::string& filter_json,
+                  std::function<void(std::string, util::Optional<AppError>)> completion_block);
+    
+    /// Runs an aggregation framework pipeline against this collection.
+    /// @param pipline A `Document` array made up of jsons strings containing the pipeline of aggregation operations to perform.
+    /// @param completion_block The resulting json string or error if one occurs
+    void aggregate(std::vector<std::string> pipline,
+                   std::function<void(std::string, util::Optional<AppError>)> completion_block);
+
+    /// Counts the number of documents in this collection matching the provided filter.
+    /// @param filter_json A `Document` as a json string that should match the query.
+    /// @param options Optional `RemoteCountOptions` to use when executing the command.
+    /// @param completion_block Returns the count of the documents that matched the filter.
+    void count(const std::string& filter_json,
+               util::Optional<RemoteFindOptions> options,
+               std::function<void(u_int64_t, util::Optional<AppError>)> completion_block);
```

Comment:
please use `uint64_t` everywhere as that is part of the standard type system (also for consistency), the `u_` versions are non-standard system defined types

Reply:
Updated to use uint64_t